src/bootstrap/src/core/build_steps/llvm.rs

```diff
@@ -109,9 +106,16 @@ pub fn prebuilt_llvm_config(builder: &Builder<'_>, target: TargetSelection) -> L
         }
     }
 
-    // Initialize the llvm submodule if not initialized already.
-    // If submodules are disabled, this does nothing.
-    builder.config.update_submodule("src/llvm-project");
+    // Although we check if we are using CI LLVM above, certain commands (like `x test mir-opt --bless`)
+    // may call this function with different targets, which could bypass the condition above. Therefore,
+    // we need to check again whether we are not using CI LLVM before fetching the LLVM submodule.
+    //
+    // Note: `Build::update_existing_submodule` is not used intentionally as we want to fetch the submodule
+    // for non-ci LLVM cases.
+    if !builder.config.llvm_from_ci {
```

Comment:
Here's a clearer version:

I'm wondering if we should add a boolean flag as an extra argument (something like `download_submodule`) and use that here instead of relying on `llvm_from_ci`. I have a feeling this could otherwise lead to some false-positive results in some edge use cases..